monitor/extensions/csv-datastore-wizard/src/main/java/org/datacleaner/monitor/server/wizard/CsvDatastoreWizardSession.java

```diff
@@ -86,26 +88,31 @@ protected WizardPageController nextPageControllerExisting() {
                 return new CsvDatastoreLocationWizardPage(getWizardContext(), "my_file.csv", false) {
 
                     @Override
-                    protected WizardPageController nextPageController(String filepath, File file) {
-                        if (!filepath.toLowerCase().endsWith(".csv")) {
-                            if (!filepath.toLowerCase().endsWith(".tsv")) {
-                                if (!filepath.toLowerCase().endsWith(".txt")) {
-                                    // only .csv and .tsv files are allowed to
-                                    // be referenced on the server, for security
-                                    // reasons.
-                                    throw new DCUserInputException(
-                                            "For security reasons, only existing .csv, .tsv or .txt files can be referenced on the server");
+                    protected WizardPageController nextPageController(Resource resource) {
+                        if (resource instanceof FileResource) {
+                            final File file = ((FileResource) resource).getFile();
+                            final String filepath = file.getAbsolutePath();
+                            if (!filepath.toLowerCase().endsWith(".csv")) {
```

Comment:
That's a pretty nasty construct. Can we use ORing instead (I'm also pretty convinced we have are tools for checking extensions, maybe they can take varargs or an array of extensions to check for)?

Why the special case for txt files?